types/dom-chromium-ai/index.d.ts

```diff
@@ -0,0 +1,249 @@
+interface WindowOrWorkerGlobalScope {
+    readonly ai: AI;
+}
+
+interface AI {
+    readonly assistant: AIAssistantFactory;
+    readonly summarizer: AISummarizerFactory;
+    readonly writer: AIWriterFactory;
+    readonly rewriter: AIRewriterFactory;
+}
+
+interface AICreateMonitor {
+    ondownloadprogress: (event: DownloadProgressEvent) => void;
+
+    addEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+}
+
+interface DownloadProgressEvent extends Event {
+    readonly loaded: number;
+    readonly total: number;
+}
+
+type AICreateMonitorCallback = (monitor: AICreateMonitor) => void;
+
+type AICapabilityAvailability = "readily" | "after-download" | "no";
+
+// Assistant
+// https://github.com/explainers-by-googlers/prompt-api/#full-api-surface-in-web-idl
+
+interface AIAssistantFactory {
+    create(options?: AIAssistantCreateOptions): Promise<AIAssistant>;
+    capabilities(): Promise<AIAssistantCapabilities>;
+}
+
+interface AIAssistantCreateOptions {
+    signal?: AbortSignal;
+    monitor?: AICreateMonitorCallback;
+
+    systemPrompt?: string;
+    initialPrompts?: AIAssistantPrompt[];
+    topK?: number;
+    temperature?: number;
+}
+
+type AIAssistantPromptRole = "system" | "user" | "assistant";
+
+interface AIAssistantPrompt {
+    role?: AIAssistantPromptRole;
+    content?: string;
+}
+
+interface AIAssistant {
+    prompt(input: string, options?: AIAssistantPromptOptions): Promise<string>;
+    promptStreaming(input: string, options?: AIAssistantPromptOptions): ReadableStream<string>;
```

Comment:
FYI: https://github.com/explainers-by-googlers/prompt-api/issues/28#issue-2458673326 has some discussion around whether this should be `AIAssistantPrompt` instead of simple `string`.